app/models/tag.rb

```diff
@@ -108,6 +108,10 @@ def tagged_documents
     @_tagged_documents ||= TaggedDocuments.new(self)
   end
 
+  def tagged_document_for_base_path(base_path)
+    tagged_documents.find { |document| document.base_path == base_path }
```

Comment:
Yeah, they should be equivalent. @Davidslv perhaps you're thinking of `select` / `filter`?